includes/compat/subscriptions/trait-wc-payment-gateway-wcpay-subscriptions.php

```diff
@@ -211,7 +215,7 @@ public function prepare_intent_for_order_pay_page(): bool {
 			return false;
 		}
 
-		$js_config                     = $this->get_payment_fields_js_config();
+		$js_config                     = WC_Payments::get_wc_payments_checkout()->get_payment_fields_js_config();
```

Comment:
Could something go wrong here? I _think_ it should be fine, because I believe the only gateway-specific function is triggered after a redirection and not on initial Order Pay page load...but probably something to keep an eye on nonetheless.